src/main.ts

```diff
@@ -14,6 +42,15 @@ export default class CheckboxReorderPlugin extends Plugin {
 		});
 	}
 
+	async loadSettings() {
+		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
+	}
+
+	async saveSettings() {
+		await this.saveData(this.settings);
+		this.reorderCheckboxes(this.app.workspace.getActiveViewOfType(MarkdownView)?.editor)
```

Comment:
You're correct. We don't want it to run _every_ time settings are changed. I think it should be ran when toggling the feature "on" however. If not, it will wait until the file is modified to update. This would cause the user to toggle the setting on, then have to make an additional modification, or run the command to re-order the checkboxes. In my opinion the expected behavior would be that it automatically reorders the checklist when the setting is toggled on. 

Let me know what you think. I can fix it.

Reply:
I mean right now it is the only setting, so it is actually ok. But better do it only if the setting is turned on